content/posts/2019-08-03---Going-from-ember-to-vue.md

```diff
@@ -71,13 +75,19 @@ The JavaScript of a component is too tightly tied to the markup (as it should be
 ## State management
 
 The popular state management solution seems to be [Vuex](https://vuex.vuejs.org/) (another "official" Vue library written by core team members).
-This is much closer to [Redux](https://redux.js.org/) or [NGRX](https://ngrx.io/).
-I'm not going to claim that these patterns are the end-all-be-all solutions for state management on the front-end, but it's where most of the community seems to be going right now.
-Which is a huge plus over Ember.
-The skills/practices you pick up writing Vuex will translate to these other libraries.
+This is much closer to [Redux](https://redux.js.org/) or [NGRX](https://ngrx.io/) -- if you're familiar with those libraries, you'll be able to jump right into Vuex.
+I'm not going to claim that these libraries are the end-all-be-all solutions for state management on the front-end, but it's where most of the community seems to be going right now.
+Which I feel is a huge plus over Ember, since the skills/practices you pick up writing Vuex will translate over to React/Redux or something similar.
 
 You are also much more explicit when updating your "store" and that makes it feel so much less magical than Ember Data.
-No, there's no identity mapping or convenient CRUD methods, but that also means you get to be more flexible with how you request data (and as a result, you'll probably be much more aware of what is going on in your application).
+In Ember data, when I call `post.save()`, it isn't immediately clear what is happening (especially as a new developer).
+It makes a `POST` request if the record is new, a `PUT` request if it is not, updates the record in our store, looks for the id being returned from the API for updates, updates our store record, etc.
+Which isn't to say that stuff is bad, but when I was first getting into Ember, I found it confusing (same with `findAll` versus `peekAll`).
+
+So no, there's no identity mapping or convenient CRUD methods, but that also means you get to be more flexible with how you request data.
+Want to use `fetch` out of the box? Go for it!
+Want to use `member-actions` or GraphQL? There's no crazy workaround here.
+As a result, you'll probably be much more aware of what is going on in your application.
```

Comment:
However, if all apps end up re-implementing the same thing, would an abstraction be warranted?

Reply:
The more I think of it, the more I should reduce this whole section to just a couple of lines and not do a straight comparison. More "Vuex is similar to redux, which is much different than state management in Ember out-of-the-box."  I feel like I could spend way too long comparing the two

Reply:
fwiw, state management in ember, out of the box, is just native javascript. (which may not be obvious until Ember Octane, where you're just setting properties on class instances).

Like with all of these frameworks/libraries, knowing regular javascript first gives you a huge edge

> I feel like I could spend way too long comparing the two

I hear ya -- when I first was planning a blog serious on react and ember, I came up with 30+ different topics :(
that's too much. lol